src/main/java/com/streamr/client/Subscription.java

```diff
@@ -154,4 +154,10 @@ private boolean checkForGap(MessageRef prev, String key) {
         MessageRef last = lastReceivedMsgRef.get(key);
         return last != null && prev.compareTo(last) > 0;
     }
+
+    public void resendDone() {
+        if (!this.resending) {
```

Comment:
Hmmm.. is this check necessary? Is it expected behavior for  `this.resending == true` or is t something exceptional? If it is exceptional an error should probably be logged.